src/averageAges.js

```diff
@@ -54,8 +68,32 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    person.mother !== null
+    && (!onlyWithSon || person.sex === 'm')
+  ));
+
+  let amountOfMothers = 0;
+
+  const sumOfAges = children.reduce((sum, person) => {
+    const mother = people.find(personOfPeople => (
```

Comment:
Good solution. I suppose you can avoid `if` statement here. Just make mother search logic before counting total age and you will be able to simplify reduce method.🙂

Reply:
Good tip! Please firstly find all people with mother and after use reduce method for calculate average age